realsense/scene_perception/win/scene_perception_object.cc

```diff
@@ -1540,6 +1544,26 @@ void ScenePerceptionObject::OnGetVoxelSize(
       scene_perception_->QueryVoxelSize(), std::string()));
 }
 
+void ScenePerceptionObject::OnGetInternalCameraIntrinsics(
+    scoped_ptr<XWalkExtensionFunctionInfo> info) {
+  Intrinsics intrinsics;
+  if (state_ == IDLE) {
+    info->PostResult(GetInternalCameraIntrinsics::Results::Create(
+        intrinsics,
+        std::string("Wrong state, start the process first.")));
+    return;  // wrong state.
+  }
+  intrinsics.image_size.width = sp_intrinsics_.imageSize.width;
+  intrinsics.image_size.height = sp_intrinsics_.imageSize.height;
+  intrinsics.focal_length.x = sp_intrinsics_.focalLength.x;
+  intrinsics.focal_length.y = sp_intrinsics_.focalLength.y;
+  intrinsics.principal_point.x = sp_intrinsics_.principalPoint.x;
+  intrinsics.principal_point.y = sp_intrinsics_.principalPoint.y;
+
+  info->PostResult(GetInternalCameraIntrinsics::Results::Create(
+      intrinsics, std::string()));
```

Comment:
I'm not using the BMI, because the data is not big, just 6 number, I use the message passing in which case we don't need to unwrap the result in the jsStub.